packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -701,7 +756,34 @@ private String getResizedImagePath(String path, @NonNull ImageSelectionOptions o
         outputOptions.getQuality().intValue());
   }
 
-  void handleVideoResult(String path) {
+  private void handleMediaResult(ArrayList<String> paths, boolean shouldDeleteOriginalIfScaled) {
+    ImageSelectionOptions localImageOptions = null;
+    synchronized (pendingCallStateLock) {
+      if (pendingCallState != null) {
+        localImageOptions = pendingCallState.imageOptions;
+      }
+    }
+
+    if (localImageOptions != null) {
+      ArrayList<String> finalPaths = new ArrayList<>();
+      for (int i = 0; i < paths.size(); i++) {
+        String finalImagePath = getResizedImagePath(paths.get(i), localImageOptions);
```

Comment:
Is this just relying on `getResizedImagePaths` failing for non-image types? If so, we should make that more explicit somehow, either by checking the type if we can, or by commenting this clearly.